### PR TITLE
feat(log): added `log_colors` configuration flag

### PR DESCRIPTION
**Pull request by cason**
Possible temporary solution for #4452.

Introduces a `log_colors` configuration parameter. Its default value is `true`, which means the currently behavior (i.e., `slog` produces colored output), but colors can be disabled by manually setting it to `false`.

In `e2e` tests, log colors are disabled by default. If you miss them, add to your manifest file:

```
config = [ "log_colors = true" ]
```

---

#### PR checklist

- [ ] Tests written/updated
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by melekes — APPROVED** [review]
awesome 👍

**Review by andynog — APPROVED** [review]
lgtm, added a few comments. Thanks!

File: test/e2e/node/main.go [hunk 1]
```diff
@@ -269,6 +269,8 @@ func setupNode() (*config.Config, log.Logger, *nodekey.NodeKey, error) {
 
 	if cmtcfg.LogFormat == config.LogFormatJSON {
 		logger = log.NewJSONLogger(os.Stdout)
+	} else if !cmtcfg.LogColors {
```
[the remark this answers is not in the document]
Comment (cason) [reply test/e2e/node/main.go:272]: I think it is implicit that JSON is "uncolored"?

I can make this `else if` a `if` too.

File: test/e2e/runner/setup.go [hunk 2]
```diff
@@ -350,6 +350,7 @@ func MakeConfig(node *e2e.Node) (*config.Config, error) {
 		cfg.LogLevel = node.Testnet.LogLevel
 	}
 
+	cfg.LogColors = false
```
[the remark this answers is not in the document]
Comment (cason) [reply test/e2e/runner/setup.go:353]: I am actually moving against adding more fields to the manifest, in particular fields that are part of Comet's config, from the introduction of https://github.com/cometbft/cometbft/pull/3967.

So, we can add colors by using

```
config = [ "log_colors = true" ]
```

And we can increase the consensus logging level by using:

```
config = [ "log_level = consensus:debug,*:info" ]
```

which renders the manifest's `log_level` field pretty redundant.

**Comment by cason** [comment]
Do we need a changelog entry here?

File: config/config.toml.tpl [hunk 3]
```diff
@@ -47,6 +47,9 @@ log_level = "{{ .BaseConfig.LogLevel }}"
 # Output format: 'plain' (colored text) or 'json'
```
Comment (cason) [inline config/config.toml.tpl:47]:
```suggestion
# Output format: 'plain' or 'json'
```

File: docs/references/config/config.toml.md [hunk 4]
```diff
@@ -271,6 +271,21 @@ log_format = "plain"
 
 <!--- Todo: Probably we should create separate schemas for the different log levels or modules. --->
 
+### log_colors
+
+Define whether the log output should be colored.
```
Comment (cason) [inline docs/references/config/config.toml.md:276]:
```suggestion
Define whether the log output should be colored.
Only relevant when [`log_format`](#log_format) is `plain`.
```

File: docs/references/config/config.toml.md [hunk 5]
```diff
@@ -271,6 +271,21 @@ log_format = "plain"
...
+```toml
+log_colors = true
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `true`  |
+|                     | `false` |
+
+The default is `true`.
```
Comment (cason) [inline docs/references/config/config.toml.md:287]:
```suggestion
The default is `true` when [`log_format`](#log_format) is `plain`.
```

File: config/config.go [hunk 6]
```diff
@@ -255,6 +255,9 @@ type BaseConfig struct {
 	// Output format: 'plain' (colored text) or 'json'
```
Comment (cason) [inline config/config.go:255]:
```suggestion
	// Output format: 'plain' or 'json'
```

File: config/config.go [hunk 7]
```diff
@@ -255,6 +255,9 @@ type BaseConfig struct {
...
 	LogFormat string `mapstructure:"log_format"`
 
+	// Colored log output
```
Comment (cason) [inline config/config.go:258]:
```suggestion
	// Colored log output when LogFormat is `plain`
```

File: test/e2e/node/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cason) [reply test/e2e/node/main.go:272]: Fixed the documentation, please check last commit.

[the remark this answers is not in the document]
Comment (andynog) [reply test/e2e/node/main.go:272]: that's more clear now on the documentation, thanks

File: test/e2e/runner/setup.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (andynog) [reply test/e2e/runner/setup.go:353]: yes we can do through the config option in the manifest. That's all good.

**Review by andynog — APPROVED** [review]
lgtm, thanks for addressing the suggestions @cason

**Pull request merged**